replica/table_check.cc

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright (C) 2024-present ScyllaDB
+ */
+
+/*
+ * SPDX-License-Identifier: AGPL-3.0-or-later
+ */
+
+#include "replica/database.hh"
+#include "replica/table_check.hh"
+#include "service/migration_manager.hh"
+
+namespace replica {
+
+future<table_dropped> table_sync_and_check(replica::database& db, service::migration_manager& mm, const table_id& uuid) {
+    bool use_raft = !mm._enable_schema_pulls;
+    if (use_raft) {
+        abort_on_expiry aoe(lowres_clock::now() + std::chrono::seconds{10});
+        auto& as = aoe.abort_source();
+        auto sub = mm._as.subscribe([&as] () noexcept {
+            if (!as.abort_requested()) {
+                as.request_abort();
+            }
+        });
+
+        // Trigger read barrier to synchronize schema.
+        co_await mm._group0_barrier.trigger(as);
```

Comment:
Why not just use `mm._as` directly here?

Reply:
To abort the barrier not only on shutdown, but also after some timeout in case when half of a cluster is down (@kbr-scylla advised me so). I'm not sure if 10s is the right timeout though